.github/workflows/ci.yml

```diff
@@ -0,0 +1,438 @@
+      - php-type-matrix
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        alpine: ${{ fromJson(needs.supported-alpine-versions.outputs.alpine) }}
+        php: ${{ fromJson(needs.supported-php-versions.outputs.php) }}
+        type: ${{ fromJson(needs.php-type-matrix.outputs.type) }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Download Images
+        uses: actions/download-artifact@v2
+        with:
+          name: docker-image-${{ matrix.type }}-${{ matrix.php }}-${{ matrix.alpine }}
+          path: ./tmp
+      - run: docker load --input ./tmp/image*.tar
+      - run: make test-${{ matrix.type }}
+  test-http:
+    name: Functionaly test nginx ${{ matrix.nginx }} with PHP FPM ${{ matrix.php }} on Alpine ${{ matrix.alpine }}
+    needs:
+      - build-http
+      - build-php
+      - build-prometheus-exporter-file
+      - supported-alpine-versions
+      - supported-nginx-versions
+      - supported-php-versions
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        alpine: ${{ fromJson(needs.supported-alpine-versions.outputs.alpine) }}
+        nginx: ${{ fromJson(needs.supported-nginx-versions.outputs.nginx) }}
+        php: ${{ fromJson(needs.supported-php-versions.outputs.php) }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Download PHP Images
+        uses: actions/download-artifact@v2
+        with:
+          name: docker-image-fpm-${{ matrix.php }}-${{ matrix.alpine }}
+          path: ./tmp
+      - name: Download nginx Images
+        uses: actions/download-artifact@v2
+        with:
+          name: docker-image-http-${{ matrix.nginx }}
+          path: ./tmp
+      - run: docker load --input ./tmp/image-fpm-${{ matrix.php }}-${{ matrix.alpine }}.tar
+      - run: docker load --input ./tmp/image-http-${{ matrix.nginx }}.tar
+      - run: sudo chown -R 1000:1000 ./test/functional/web/tmp/ # Ensure we have the same uid:gid as our `app` docker user
+        shell: bash
+      - run: make test-http
+      - run: make test-http-e2e
+  test-prometheus-exporter-file:
+    name: Functionaly test prometheus-exporter-file
+    needs:
+      - build-http
+      - build-php
+      - build-prometheus-exporter-file
+      - supported-nginx-versions
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+      - name: Download nginx Images
+        uses: actions/download-artifact@v2
+        with:
+          name: docker-image-prometheus-exporter-file
+          path: ./tmp
+      - run: docker load --input ./tmp/image-prometheus-exporter-file.tar
+      - run: sudo chown -R 1000:1000 ./test/functional/web/tmp/ # Ensure we have the same uid:gid as our `app` docker user
+        shell: bash
+      - run: make test-prometheus-exporter-file-e2e
+  check-mark: # This is our required step, pay extra attention when this step is changed for what ever reason
```

Comment:
what is this for?

Reply:
We have branch protection on the main branch, and instead of having to check/uncheck 60 required checks when we add/remove new PHP versions we only make this job required as it requires the previous jobs to pass before it even runs. This way we don't have to update the required checks when a job's name changes or a new version comes into play, or when we consolidate all the build jobs into one.

Reply:
ah, i suggest using https://github.com/probot/settings for managing that stuff, not manually clicking in the UI. HTH!

Reply:
That could work, but it would mean updating the settings file and the naming/versions in two different PR as AFAIK the settings need to be applied/merged before they are effective on the PR. (I do love that suggestions, using it for my personal repo's to managed settings :+1: .)

Reply:
that is true - but we do the update in one PR and just have an admin merge :)

Reply:
That is also a possibility. But what I would like even more is that they let us select the underlying job name instead of the fancy user friendly name, that way you only have to select a handful instead of nearly a 100 :zipper_mouth_face: . brb doing a feature suggestion to GitHub for that